datenguide_python/query_builder.py

```diff
@@ -20,32 +25,61 @@ class Field(NamedTuple):
         If the filter is not set, then only the summed result is returned.)
     """
 
-    name: str
-    subfields: list
-    args: Optional[Dict[str, Union[str, List[str]]]] = None
+    def __init__(
+        self,
+        name: str,
+        fields: List[Union[str, object]] = [],
+        args: Optional[Dict[str, Union[str, List[str]]]] = None,
+    ):
+
+        self.name = name
+        self.fields = fields
+        self.args = args
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def add_field(self, field: Union[str, "Field"]) -> object:
+        if isinstance(field, str):
+            added_field = Field(name=field)
+        else:
+            added_field = field
+
+        if self.fields:
+            self.fields.append(added_field)
+        else:
+            self.fields = [added_field]
+        return added_field
+
+    def add_args(self, args: dict):
+        if self.args:
+            self.args.update(args)
+        else:
+            self.args = args
+
+    def get_info(self) -> Optional[TypeMetaData]:
+        return QueryExecutioner().get_type_info(self.name)
 
 
-class QueryBuilder:
+class Query:
     def __init__(
         self,
-        fields: List[Union[str, Field]],
+        fields: List[Union[str, Field]] = [],
         region: str = None,
```

Comment:
We might want to consider to have factory methods instead of arguments for all types of queries here. I'm thinking of something like the following.
```python
class Query:
    def __init__(self,start_field):
        ...

    @classmethod
    def regionQuery(cls,regionid):
        return cls.__init__(start_field = Field("region",[],args = {'id':regionid}))

    @classmethod
    def allRegionsQuery(cls):
        return cls.__init__(start_field = Field("allRegions",[]))

   def add_field(...):
       self.start_field.add_field(...)
```

In this context it might then also make sense to move some of the recursive helper Field functionality to the Field class and simply call it in the Query class from start_field.